uiowa/uiowa
# [Classrooms] building numbers, link to facilities

Resolves #6592 





# How to test



```
git checkout classrooms-building-numbers && git pull && ddev blt ds --site=classrooms.uiowa.edu && ddev drush @classrooms.local uli /admin/structure/building
```
Check for a clean sync
Spot check some buildings and see that they have building numbers
Check that the following buildings aren't available for selection in creating rooms, filtering searches, etc.
- Calvin Hall
- South Quad
- Pharmaceutical Sciences Research Building
- Kirkwood Regional Center

```
ddev drush @classrooms.local classrooms-buildings
```
No new buildings should be imported (assuming the api results haven't changed by testing time)

Delete a building or multiple buildings
```
ddev drush @classrooms.local classrooms-buildings
```
The message should indicate that the same number of buildings you deleted should now be created 
Check the building(s) that you'd deleted, and see that they have been created, including their building number

Check some Room nodes, and see that the Building (toward the bottom of the display) is linked, and that link points to the correct building page on facilities.uiowa.edu.

Check that the Room teaser card building links are also correct (such as on https://classrooms.uiowa.ddev.site/find-university-classroom)


This work moves from the previous endpoint https://api.maui.uiowa.edu/maui/api/pub/registrar/courses/AstraBldgRmCompleteList/list with a custom filter to using https://api.maui.uiowa.edu/maui/api/pub/facilityBuildingRoom/list?roomCategory=UNIVERSITY_CLASSROOM and https://api.maui.uiowa.edu/maui/api/pub/facilityBuildingRoom/list?roomCategory=PROGRAMMED_CLASSROOM.
The previous filters were on the room "RegionList" property in the full list, filtered down to only:
```
        '1) University Classrooms - Level 1',
        '1) University Classrooms - Original',
        '1) University Classrooms',
        '1) University Classrooms - Study Space',
        '1) Programmed Classrooms - Level 2',
        'Classroom-Programmed',
```
As far as I can tell, we're getting the same end results with this switch (comparison of resultant buildings of the building process, as well as spot checking rooms contained between the two lists), but any extra checking here, or other methods of comparison are welcome.

I thought of writing a script to do our old filtering and compare the results with a full pull of the other two APIs for a thorough comparison, but haven't gone that far yet. It could be an option, though.

## Other changes

The RoomSchedule class was updated to use dependency injection for its services.
Spot check some rooms and compare their schedules with the room on classrooms.prod.drupal.uiowa.edu to spot check for regressions. (Largely this should be a check that it still functions--the services affected are those to get the current node form the path, to load the maui api, and to create a pager, so it's pretty much a "this still works" or "this is all broken" type update)
